python/lib/core/dmod/core/dataset.py

```diff
@@ -728,10 +816,12 @@ def link_user(self, user: DatasetUser, dataset: Dataset) -> bool:
             Whether the link was successful.
         """
         if dataset.name not in self.datasets:
-            raise RuntimeError("Cannot link user {} to unknown dataset {}".format(user.uuid, dataset.name))
-        if dataset.name not in self._dataset_users:
-            self._dataset_users[dataset.name] = set()
-        self._dataset_users[dataset.name].add(user.uuid)
+            raise RuntimeError(f"Cannot link user {user.uuid!s} to unknown dataset {dataset.name}")
```

Comment:
Partially because I look at this as expecting a dataset that isn't there, which feels more like an error than a valid completion of the function.

Partially because I want to pass back some information on why things didn't work, though I suppose I could plug things into a result indicator to do that.

I'll also note that returning a `bool` value was more significant at earlier points of writing this, and then just left in and not thought about.  It is still kind of needed for `DatasetUser.link_to_dataset`, though that could be adjusted easily.

Reply:
Yeah that is fair. This is how you are handling errors in other nearby places too. I am not the biggest fan of the only result from the method being `True`, but I will let you have the final say on if you want to change it or not.

Reply:
For now, I am going to leave this as it is.  We may want to come back to such things at the same time we deal with #418.